src/app/components/lineage-graph/lineage-graph.component.ts

```diff
@@ -63,4 +63,20 @@ export class LineageGraphComponent {
     public onClickInfo(): void {
         this.showSidePanel = !this.showSidePanel;
     }
+
+    private lineageGraphHeight(): number {
+        const headerViewMenuElement = document.getElementById("app-dataset-view-header");
+        const datasetTabsElement = document.getElementById("container-dataset-tabs");
+        const headerElement = document.getElementById("app-header");
+        if (headerViewMenuElement && datasetTabsElement && headerElement) {
+            const headerHeight = headerElement.getBoundingClientRect().height;
+            const headerViewHeight = headerViewMenuElement.getBoundingClientRect().height;
+            const headerMarginTop = +getComputedStyle(headerViewMenuElement).marginTop.split("px")[0];
+            const datasetTabsElementHeight = datasetTabsElement.getBoundingClientRect().height;
+            return (
+                window.innerHeight - (headerHeight + headerViewHeight + headerMarginTop + 2 * datasetTabsElementHeight)
+            );
+        }
+        return screen.height - 390;
```

Comment:
It's architecturally incorrect for middle-layer component to assume a certain environment in higher level components in the hierarchy. It should be getting the computed number through @Input port. Only the main component `AppComponent` is aware of where header exists. Only `DatasetComponent` knows about dataset tabs. `LineageGraphComponent` is not allowed to access this information.

Reply:
Plz report a separate ticket for this problem, and revert to constant for now.